rotkehlchen/globaldb/handler.py

```diff
@@ -487,51 +485,47 @@ def get_assets_mappings(identifiers: list[str]) -> tuple[dict[str, dict], dict[s
 
     @staticmethod
-    def search_assets(
-            filter_query: 'AssetsFilterQuery',
-            db: 'DBHandler',
-    ) -> list[dict[str, Any]]:
+    def search_assets(filter_query: 'AssetsFilterQuery', db: 'DBHandler') -> list[dict[str, Any]]:
         """Returns a list of asset details that match the search query provided."""
         search_result = []
-        globaldb = GlobalDBHandler()
-        query, bindings = filter_query.prepare()
-        query = ALL_ASSETS_TABLES_QUERY.format(dbprefix='globaldb.') + query
-        resolved_eth = A_ETH.resolve_to_crypto_asset()
-        globaldb._wal_checkpoint()
+        should_skip = filter_query.ignored_assets_handling.get_should_skip_handler()
         with db.conn.read_ctx() as cursor:
             treat_eth2_as_eth = db.get_settings(cursor).treat_eth2_as_eth
-            cursor.execute(
-                f'ATTACH DATABASE "{globaldb.filepath()!s}" AS globaldb KEY "";',
-            )
-            try:
-                cursor.execute(query, bindings)
-                found_eth = False
-                for entry in cursor:
-                    if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa: E501
-                        if found_eth is False:
-                            search_result.append({
-                                'identifier': resolved_eth.identifier,
-                                'name': resolved_eth.name,
-                                'symbol': resolved_eth.symbol,
-                                'is_custom_asset': False,
-                            })
-                            found_eth = True
-                        continue
-
-                    entry_info = {
-                        'identifier': entry[0],
-                        'name': entry[1],
-                        'symbol': entry[2],
-                        'is_custom_asset': AssetType.deserialize_from_db(entry[4]) == AssetType.CUSTOM_ASSET,  # noqa: E501
-                    }
-                    if entry[3] is not None:
-                        entry_info['evm_chain'] = ChainID.deserialize_from_db(entry[3]).to_name()
-                    if entry[5] is not None:
-                        entry_info['custom_asset_type'] = entry[5]
-
-                    search_result.append(entry_info)
-            finally:
-                cursor.execute('DETACH DATABASE globaldb;')
+            ignored_assets = db.get_ignored_asset_ids(cursor)
+
+        query, bindings = filter_query.prepare(without_ignored_asset_filter=True)
+        query = ALL_ASSETS_TABLES_QUERY + query
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            found_eth = False
+            for entry in cursor:
+                if should_skip(entry[0], ignored_assets):
+                    continue
+
+                if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):
```

Comment:
Also isn't; there a lot of common functionality in the parts of the code you touched that can be abstracted? Or is it just me?

Reply:
so they seems similar because only these are common in all the functions: ignored_assets and should_skip that I already abstracted. Other things are slightly different in all of them.